### ISSUE TITLE
ETagMessageHandler.CreateETag method has a serious bug

### ISSUE THREAD

**Issue by wei-zhou** [OData/WebApi#496]
The source code is as below.

``` C#
IEdmModel model = entityInstanceContext.EdmModel; 
IEdmEntitySet entitySet = entityInstanceContext.NavigationSource as IEdmEntitySet; 
IEnumerable<IEdmStructuralProperty> concurrencyProperties; 
if (model != null && entitySet != null) 
{ 
    concurrencyProperties = model.GetConcurrencyProperties(entitySet).OrderBy(c => c.Name); 
} 
else 
{ 
    concurrencyProperties = Enumerable.Empty<IEdmStructuralProperty>(); 
}
```

If the **navigation source** is a **contained entity set**, the **ETag** will become **null**. I think this is  NOT a right behavior. But it is right in the last release.

**Comment by congysu** [comment]
Need to change ODataLib's set ETag API:
https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs#L1076

Need to confirm `EntitySet` includes contained entity set or not.
```xml
<Term Name="OptimisticConcurrency" Type="Collection(Edm.PropertyPath)" AppliesTo="EntitySet">
<Annotation Term="Core.Description" String="Data modification requires the use of Etags. A non-empty collection contains the set of properties that are used to compute the ETag"/>
</Term>
```

**Comment by mikepizzo** [comment]
Fixed in PR #951.
